ui/src/text_annotator.tsx

```diff
@@ -227,22 +260,47 @@ export
             <span style={removeIconStyle as React.CSSProperties} className={css.iconUnderlay}></span>
           </mark>
         )
-      }
+      },
+      handleMouseDown = (startElProps: TokenMouseEventProps) => () => {
+        startElPropsRef.current = startElProps
+      },
+      handleMouseUp = (endElProps: TokenMouseEventProps) => () => {
+        annotate(endElProps)
+      },
+      handleMouseLeave = (endElProps: TokenMouseEventProps) => (ev: React.MouseEvent<HTMLSpanElement>) => {
+        if (startElPropsRef.current !== undefined && (ev.relatedTarget as any)?.nodeName !== 'P') annotate(endElProps) // nodeName prop is not typed yet
+      },
+      text = tokens.map((token, idx) => {
+        return <Token
+          key={idx}
+          idx={idx}
+          tokenProps={token}
+          handleMouseDown={handleMouseDown}
+          handleMouseUp={handleMouseUp}
+          handleMouseLeave={handleMouseLeave}
+          getMark={getMark}
+          activeColor={activeTag ? tagColorMap.get(activeTag)! : undefined}
+        />
+      }, [] as React.ReactElement[])
 
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    React.useEffect(() => { wave.args[model.name] = model.items as unknown as Rec[] }, [])
+    React.useEffect(() => { wave.args[name] = items as unknown as Rec[] }, [])
+
 
     return (
-      <div data-test={model.name} className={model.readonly ? css.readonly : ''}>
-        <div className={clas('wave-s16 wave-w6', css.title)}>{model.title}</div>
-        <AnnotatorTags tags={model.tags} activateTag={activateTag} activeTag={activeTag} />
-        <div className={clas(css.content, 'wave-s16 wave-t7 wave-w3')}>
-          {
-            tokens.map(({ text, tag }, idx) => (
-              <span key={idx} onMouseDown={updateStartIdx(idx)} onMouseUp={annotate(idx)}>{tag ? getMark(text, idx, tag) : text}</span>
-            ))
-          }
-        </div>
+      <div data-test={name} className={readonly ? css.readonly : ''}>
+        <div className={clas('wave-s16 wave-w6', css.title)}>{title}</div>
+        <AnnotatorTags tags={tags} activateTag={activateTag} activeTag={activeTag} />
+        <Fluent.Toggle
+          data-test={'toggle'}
+          label={'Smart selection'}
+          defaultChecked={smartSelection}
+          onChange={() => setSmartSelection(!smartSelection)}
```

Comment:
Make sure to always get the most recent value.
```suggestion
          onChange={() => setSmartSelection(prevSelection => !prevSelection)}
```